bms-monitor.js

```diff
@@ -1,18 +1,56 @@
-const {expect} = require('chai');
+import { expect } from 'chai';
+import { minTemperatureRange, maxTemperatureRange, minsocRange, maxsocRange, mincharge_rateRange } from './range.constants.js';
+import { showMessage } from './showMessage.js';
+import { checkTempLimit, checkSocLimit, checkChargeLimit } from './checkMinMaxLimit.js'
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
-    }
-    return true;
+function batteryIsOk(funcationName, minValue, maxValue, currentValue, range, type) {
+    return {
+        temperature: function () {
+            
+            showMessage(range, currentValue, type);
+            return checkTempLimit(minValue, maxValue, currentValue);
+        },
+        soc: function () {
+            showMessage(range, currentValue, type);
+            return checkSocLimit(minValue, maxValue, currentValue);
+        },
+        charge_rate: function () {
+            showMessage(range, currentValue, type);
+            return checkChargeLimit(maxValue, currentValue);
+        }
+    }[funcationName]
 }
 
-expect(batteryIsOk(25, 70, 0.7)).to.be.true;
-expect(batteryIsOk(50, 85, 0)).to.be.false;
+const socRange = {
+    'LOW_BREACH': [0, 20],
+    'LOW__WARNING': [21, 24],
+    'NORMAL': [24, 75],
+    'HIGH_WARNING': [76, 80],
+    'HIGH_BREACH': [81, 100]
+}
+
+const chargeRange = {
+    'NORMAL': [0.4, 0.5],
+    'HIGH_WARNING': [0.6, 0.68],
+    'HIGH_BREACH': [0.7, 0.8]
+}
+
+const temperatureRange = {
+    'LOW_BREACH': [0, 10],
+    'LOW__WARNING': [11, 20],
+    'NORMAL': [21, 30],
+    'HIGH_WARNING': [31, 35],
+    'HIGH_BREACH': [36, 40]
+}
+
+const checkSoc = batteryIsOk('soc', minsocRange, maxsocRange, 70, socRange, 'soc' );

```

Comment:
The name batteryIsOk is used to check individual parameter. Maybe BatteryParameterIsOk could have been a better name. Please think over this.

Now each parameter is checked for range. The API is such that the user has to check each parameter and then decide whether battery is ok or not.